pipeline_components/aiplatform/aiplatform/custom_train_job/component.py

```diff
@@ -0,0 +1,167 @@
+# Copyright 2022 Google LLC
+from typing import NamedTuple, List, Dict
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     https://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from kfp.v2.dsl import Input, component, Metrics, Output, Artifact, Dataset
+from pathlib import Path
+
+
+@component(
+    base_image="python:3.7",
+    packages_to_install=["google-cloud-aiplatform==1.24.1"],
+    output_component_file=str(Path(__file__).with_suffix(".yaml")),
+)
+def custom_train_job(
+    task_uri: str,
```

Comment:
suggest to rename `task` to `train` or `training_script` everywhere to avoid confusion with pipeline tasks